### PR TITLE
Added an API to fetch all ratings by a User on different Skills

### LINKED ISSUE #807: Create API to fetch all the ratings the user has given to skills
**Actual Behaviour**

Currently, there is no such API.

**Expected Behaviour**

An API should be implemented to fetch all the ratings the user has given to skills from `data/skill_rating/fiveStarSkillRating.json`

**Would you like to work on the issue?**

Yes

### PR DESCRIPTION
**Pull request by Akshat-Jain**
Fixes #807

Changes: Added an API to fetch all ratings by a User on different Skills.

For example, for the following `fiveStarSkillRating.json` file contents : 
```
{"general": {
  "Shopping": {"en": {"amazon_shopping": [
    {
      "stars": 1,
      "email": "akjn11@gmail.com",
      "timestamp": "2018-06-10 13:05:32.295"
    },
    {
      "stars": 3,
      "email": "akjn22@gmail.com",
      "timestamp": "2018-06-10 13:05:32.295"
    }
  ]}},
  "Knowledge": {"en": {
    "aboutsusi": [
      {
        "stars": 2,
        "email": "akjn11@gmail.com",
        "timestamp": "2018-06-10 13:26:26.222"
      }
    ],
    "acronyms": [{
      "stars": 4,
      "email": "akjn22@gmail.com",
      "timestamp": "2018-06-10 13:03:00.134"
    }],
    "anagrams": [{
      "stars": 3,
      "email": "akjn11@gmail.com",
      "timestamp": "2018-06-10 13:25:31.195"
    }]
  }}
}}
```

On passing access_token for akjn11@gmail.com as a query parameter, we get : 
```
{
  "rated_skills": [
    {"amazon_shopping": {
      "stars": "1",
      "timestamp": "2018-06-10 13:05:32.295"
    }},
    {"aboutsusi": {
      "stars": "2",
      "timestamp": "2018-06-10 13:26:26.222"
    }},
    {"anagrams": {
      "stars": "3",
      "timestamp": "2018-06-10 13:25:31.195"
    }}
  ],
  "session": {"identity": {
    "type": "email",
    "name": "akjn11@gmail.com",
    "anonymous": false
  }},
  "accepted": true,
  "message": "User ratings fetched."
}
```

Also handles the case when User hasn't rated any skills yet : 
```
{
  "session": {"identity": {
    "type": "email",
    "name": "akjn88@gmail.com",
    "anonymous": false
  }},
  "accepted": true,
  "message": "User has not rated any Skills yet."
}
```

### CODE REVIEW

File: src/ai/susi/server/api/cms/ProfileDetailsService.java [hunk 1]
```diff
@@ -0,0 +1,156 @@
...
+
+    @Override
+    public String getAPIPath() {
+        return "/cms/getProfileDetails.json";
+    }
+
+    @Override
+    public ServiceResponse serviceImpl(Query query, HttpServletResponse response, Authorization authorization, final JsonObjectWithDefault permissions) throws APIException {
+
+    String email = authorization.getIdentity().getName();
```
Comment (akshatnitd) [inline src/ai/susi/server/api/cms/ProfileDetailsService.java]: Set email after null check in line 80

File: src/ai/susi/server/api/cms/ProfileDetailsService.java [hunk 2]
```diff
@@ -0,0 +1,156 @@
...
+    JSONObject groupObject = new JSONObject();
+    JSONObject languageObject = new JSONObject();
+    JSONArray skillnameArray = new JSONArray();
+    JSONArray skillRating = new JSONArray();
+
+    if (authorization.getIdentity() == null) {
+        throw new APIException(400, "Specified user data not found, ensure you are logged in");
+    }
+
+    for(String model_name:fiveStarSkillRating.keys())
```
Comment (akshatnitd) [inline src/ai/susi/server/api/cms/ProfileDetailsService.java]: Add space before and after `:`

File: src/ai/susi/server/api/cms/ProfileDetailsService.java [hunk 3]
```diff
@@ -0,0 +1,156 @@
...
+        while(groupNames.hasNext()) {
+            String key = (String) groupNames.next();
+            groupnameKeysList.add(key);
+        }
+        
+        for(String group_name : groupnameKeysList)
+        {
+            groupObject = modelObject.getJSONObject(group_name);
+            Iterator languageNames = groupObject.keys();
+
```
Comment (akshatnitd) [inline src/ai/susi/server/api/cms/ProfileDetailsService.java]: Remove extra new lines where not required

File: src/ai/susi/server/api/cms/ProfileDetailsService.java [hunk 4]
```diff
@@ -0,0 +1,156 @@
...
+            Iterator languageNames = groupObject.keys();
+            
+            List<String> languagenameKeysList = new ArrayList<String>();
+            
+            while(languageNames.hasNext()) {
+                String key = (String) languageNames.next();
+                languagenameKeysList.add(key);
+            }
+
+            for(String language_name : languagenameKeysList){
```
Comment (akshatnitd) [inline src/ai/susi/server/api/cms/ProfileDetailsService.java]: Add space before `{`

File: src/ai/susi/server/api/cms/ProfileDetailsService.java [hunk 5]
```diff
@@ -0,0 +1,156 @@
...
+                Iterator skillNames = languageObject.keys();
+                
+                List<String> skillnamesKeysList = new ArrayList<String>();
+                
+                while(skillNames.hasNext()) {
+                    String key = (String) skillNames.next();
+                    skillnamesKeysList.add(key);
+                }
+
+                for(String skill_name : skillnamesKeysList){
```
Comment (akshatnitd) [inline src/ai/susi/server/api/cms/ProfileDetailsService.java]: Add space before {

Comment (akshatnitd) [reply src/ai/susi/server/api/cms/ProfileDetailsService.java]: Make this change throughout the file

File: src/ai/susi/server/api/cms/ProfileDetailsService.java [hunk 6]
```diff
@@ -0,0 +1,156 @@
...
+                            result.put(jsonEmail, skillRating);
+                        }
+                    }
+                }
+            }
+        }
+    }
+
+    if(result.length()==0){
+        result.put("accepted", true);
```
Comment (akshatnitd) [inline src/ai/susi/server/api/cms/ProfileDetailsService.java]: Put value of `accepted` as `false` in this case

**Comment by Akshat-Jain** [comment]
@akshatnitd Made the changes. Please review. Thanks :-)

File: src/ai/susi/server/api/cms/ProfileDetailsService.java [hunk 7]
```diff
@@ -0,0 +1,155 @@
...
+                for(String skill_name : skillnamesKeysList)
+                {
+                    skillnameArray = languageObject.getJSONArray(skill_name);
+
+                    for(int i=0; i<skillnameArray.length();i++){
+                        String jsonEmail = skillnameArray.getJSONObject(i).get("email").toString();
+                        if(jsonEmail.equals(email)) {
+                            JSONObject userSkillData = new JSONObject();
+                            JSONObject userSkillRatings = new JSONObject();
+                            String stars = skillnameArray.getJSONObject(i).get("stars").toString();
```
Comment (akshatnitd) [inline src/ai/susi/server/api/cms/ProfileDetailsService.java]: Send the value of stars here as `int`

File: src/ai/susi/server/api/cms/ProfileDetailsService.java [hunk 8]
```diff
@@ -0,0 +1,155 @@
...
+                while(skillNames.hasNext()) {
+                    String key = (String) skillNames.next();
+                    skillnamesKeysList.add(key);
+                }
+
+                for(String skill_name : skillnamesKeysList)
+                {
+                    skillnameArray = languageObject.getJSONArray(skill_name);
+
+                    for(int i=0; i<skillnameArray.length();i++){
```
Comment (akshatnitd) [inline src/ai/susi/server/api/cms/ProfileDetailsService.java]: Add space before `{`

**Comment by Akshat-Jain** [comment]
@akshatnitd Made the changes. Please check. Thanks :-)

**Comment by Akshat-Jain** [comment]
@Orbiter @dynamitechetan @DravitLochan @saurabhjn76 Please review. Thanks :-)

**Comment by Akshat-Jain** [comment]
@akshatnitd Resolved the conflicts :-)
@Orbiter @dynamitechetan @DravitLochan @saurabhjn76 Please review. Thanks :-)

**Pull request merged**